implementations/python/sugar-source-tree/src/sugar_source_tree/nodes.py

```
@@ -4596,13 +4596,22 @@ def _prebound_manager_resolution(self, item: WithItem):
-                observed=str(exc),
-                requested="one contract-resolution row for every enrolled With demand",
-                fix="repair prereq-2 demand/table generation; never search at construction",
+                observed=(
+                    "no context-manager derivation for source coordinate "
+                    f"{coordinate}"
+                ),
+                requested="one resolved authenticated ContextManagerContractRefV1",
+                fix="publish or derive the exact typed CM contract before construction",
             )
+            self.reporter.report_gap(self, panic)
+            raise panic
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)